packages/vm/src/context.rs

```diff
@@ -329,11 +324,11 @@ pub(crate) fn with_iterator_from_context<'a, 'b, S, Q: 'b, F, T>(
 where
     S: Storage,
     Q: Querier,
-    F: FnOnce(&'b mut (dyn Iterator<Item = FfiResult<(KV, u64)>>)) -> VmResult<T>,
+    F: FnOnce(&'b mut (dyn StorageIterator + 'b)) -> VmResult<T>,
 {
     let b = get_context_data_mut::<S, Q>(ctx);
     match b.iterators.get_mut(&iterator_id) {
-        Some(iterator) => func(iterator),
+        Some(iterator) => func(&mut **iterator),
```

Comment:
This `&mut **` is not strictly required, but it removes one layer of indirection when `func` accesses the argument, by giving it a pointer to the inner object rather than the box.
This is totally a micro-optimization, but i thought it was cleaner this way.

Reply:
Happy to hear we don't need this change and can go back to the original code. Implemented here: 6d17dbfac6db3204d63ac6a1e3ffa6144ed27263